crates/astria-celestia-client/src/blob_space.rs

```diff
@@ -69,29 +305,42 @@ pub struct RollupNamespaceData {
 }
 
 impl RollupNamespaceData {
-    /// Verifies `self.inclusion_proof` given the chain ID and the root of the merkle tree
-    /// constructed from `self.rollup_txs` and the provided `root_hash`.
+    /// Utility to check if the rollup blob belongs to the given sequencer blob.
+    ///
+    /// Returns `Ok` If `self` belongs to `sequencer_blob`.
     ///
     /// # Errors
-    /// Returns an error if the inclusion proof could not be verified.
-    pub fn verify_inclusion_proof(
+    /// Returns an error in the following cases:
+    /// 1. the block hash of the rollup blob does not match that of `sequencer_blob`
+    /// 2. `self.chain_id` is not listed in `sequencer_blob.rollup_chain_ids`
+    /// 3. `self.inclusion_proof` could not be verified against `sequencer_blob.action_tree_root`.
+    pub fn belongs_to(
         &self,
-        root_hash: [u8; 32],
-    ) -> Result<(), RollupVerificationFailure> {
+        sequencer_blob: &SequencerNamespaceData,
+    ) -> Result<(), RollupDoesNotBelong> {
+        // XXX: The return order of the error conditions is important: `RollupNotInTree`
+        //      explicitly mentions that the Chain ID is listed in sequencer_blob, but
+        //      that its inclusion could not be verified.
```

Comment:
maybe mention that this means that the rollup transactions are corrupted?

Reply:
This note is more about the error order in case somebody wants to refactor and does not pay attention.